File: ToDoWebApi/ToDo.DAL/Models/ToDoItem.cs
using System.ComponentModel.DataAnnotations;

namespace ToDo.DAL.Models
{
    /// <summary>
    /// Model for a to-do item
    /// </summary>
    public class ToDoItem
    {
        /// <summary>
        /// Identifier
        /// </summary>
        [Key]
        public int Id { get; set; }

        /// <summary>
        /// Task description
        /// </summary>
        public string Note { get; set; }

        /// <summary>
        /// Status flag
        /// </summary>
        public bool IsCompleted { get; set; }
    }
}

File: ToDoWebApp/src/app/models/todo-item.ts
export class ToDoItem{
    public id!: number;
    public note!: string;
    public isCompleted!: boolean;

    constructor(id: number, note: string, isCompleted: boolean){
        this.id = id;
        this.note = note;
        this.isCompleted = isCompleted;
    }
}